### PR TITLE
feat: implement setup-oras

### PR DESCRIPTION
**Pull request by shizhMSFT**
Implement setup-oras as a JavaScript action

Resolves: https://github.com/oras-project/oras/issues/744

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+jobs:
+  test:
+    name: Test Setup ORAS CLI
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        version:
+          - 0.15.0
```
Comment (FeynmanZhou) [inline .github/workflows/test.yml:38]: Why do we add v0.15.0 to the test workflow?

Comment (shizhMSFT) [reply .github/workflows/test.yml:38]: It should be `v0.15.1` and it is the last version to support ORAS artifact.

Comment (shizhMSFT) [reply .github/workflows/test.yml]: Also, we usually test the latest 2 versions.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+  - run: oras version
+```
+
+Install a specific version of the `oras` CLI:
+
+```yaml
+steps:
+  - uses: oras-project/setup-oras@main
+    with:
+      version: 0.16.0
```
Comment (toddysm) [inline README.md:23]: Can we add documentation how does one request specific version of ORAS? Not only 0.16.0

**Review by toddysm — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (shizhMSFT) [reply README.md:23]: This is the documentation to request specific version of oras. I will re-write this basic documentation to a detailed one.

**Review by sabre1041 — APPROVED** [review]
Tested and works great

**Review by toddysm — APPROVED** [review]
LGTM

**Review by SteveLasker — APPROVED** [review]
Great to see the setup effort, simplifying the acquisition experience. 
LGTM

**Pull request merged**